p2p/cryptoBox/box.go

```diff
@@ -0,0 +1,159 @@
+package cryptoBox
+
+import (
+	crypto_rand "crypto/rand"
+	"errors"
+	"fmt"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"golang.org/x/crypto/nacl/box"
+	"io"
+)
+
+const (
+	keySize = 32 // non-configurable, expected by NaCl
+	nonceSize = 24 // non-configurable, expected by NaCl
+)
+
+type Key interface {
+	Bytes() []byte
+	Raw() *[keySize]byte
```

Comment:
go's arrays are already pointer, the `*` isn't really needed here. unless you have a reason ?

Reply:
If you look at the signatures of box's methods, all keys and nonces are pointers to arrays. I followed their lead as I need to pass these values in.

I'm also not sure you're right. As far as I can tell, *slices* in go are pointers, but *arrays* aren't, if you don't make them pointers explicitly.

Reply:
I guess you're right I was wrong. sorry about misleading. anyway, as @antonlerner already pointed there's no reason to store a pointer here.

Reply:
fixed: it's no longer a pointer.